src/CommunityToolkit.Maui/Converters/MathExpressionConverter/MathExpression.shared.cs

```diff
@@ -4,94 +4,179 @@
 
 namespace CommunityToolkit.Maui.Converters;
 
+enum MathTokenType
+{
+	Value,
+	Operator,
+};
+
+class MathToken
+{
+	public MathToken(MathTokenType type, string text, object? value)
+	{
+		Type = type;
+		Text = text;
+		Value = value;
+	}
+
+	public MathTokenType Type { get; }
+	public string Text { get; }
+	public object? Value { get; }
+};
+
 sealed partial class MathExpression
 {
 	const NumberStyles numberStyle = NumberStyles.Float | NumberStyles.AllowThousands;
 
 	static readonly IFormatProvider formatProvider = new CultureInfo("en-US");
 
 	readonly IReadOnlyList<MathOperator> operators;
-	readonly IReadOnlyList<double> arguments;
+	readonly IReadOnlyList<object> arguments;
+
+	internal static bool __bool(object? value)
+	{
+		if (value is null)
+		{
```

Comment:
would be great if we can make this use switch expression:

```cs
bool Bla(object b) =>
    b switch 
    {
        bool x => x,
        null => false,
        double doubleValue => doubleValue != 0 && doubleValue != double.NaN,
        string stringValue => !string.IsNullOrEmpty(stringValue),
        _ => Convert.ToBoolean(b)         
    };
    
```

Reply:
Thanks @pictos. I migrated to the new C# switch statement as per your feedback.